### PR TITLE
Fixed Managed DNS issue: added same value for element reference and m…

### PR DESCRIPTION
**Pull request by kopachevsky**
Fix for Error in community/cloud-foundation/templates/dns_managed_zone/dns_managed_zone.py

### CODE REVIEW

File: dm/templates/dns_managed_zone/dns_managed_zone.py.schema [hunk 1]
```diff
@@ -30,9 +30,12 @@ required:
 properties:
   zoneName:
     type: string
+    pattern: ^[a-z]{1}[a-z0-9-]*[a-z0-9]?$
```
Comment (bohdanyurov-gl) [inline dm/templates/dns_managed_zone/dns_managed_zone.py.schema]: This expression is incorrect (allows ending dashes). Correct one:
^[a-z]{1}([a-z0-9-]*[a-z0-9])?$

File: dm/templates/dns_managed_zone/dns_managed_zone.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def generate_config(context):
     managed_zone_name = context.properties.get('zoneName')
     dnsname = context.properties['dnsName']
     managed_zone_description = context.properties['description']
-    name_servers = '$(ref.' + context.properties['zoneName'] + '.nameServers)'
+    name_servers = '$(ref.' + context.env['name'] + '.nameServers)'
```
Comment (bohdanyurov-gl) [inline dm/templates/dns_managed_zone/dns_managed_zone.py]: What this is required for?

Comment (kopachevsky) [reply dm/templates/dns_managed_zone/dns_managed_zone.py]: resource name was 
```
 managed_zone = {	    managed_zone = {
        'name': context.env['name'],
```

But it was referenced as 
```
context.properties['zoneName'] 
```
which it not always the same

File: dm/templates/dns_managed_zone/dns_managed_zone.py.schema (hunk 1 above)
Comment (bohdanyurov-gl) [reply dm/templates/dns_managed_zone/dns_managed_zone.py.schema]: Updated example for string limit:
^[a-z]([a-z0-9-]{0,61}[a-z0-9])?$

Comment (kopachevsky) [reply dm/templates/dns_managed_zone/dns_managed_zone.py.schema]: applied

**Comment by kopachevsky** [comment]
I signed it!

**Comment by ocsig** [comment]
https://github.com/GoogleCloudPlatform/deploymentmanager-samples/issues/375

**Comment by kopachevsky** [comment]
I signed it!

**Pull request merged**